doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,317 @@
+:file:`soc.yml` file as:
+
+.. code-block:: yaml
+
+   family:
+     name: <family-name>
+     series:
+     - name: <series-1-name>
+       socs:
+       - name: <soc1>
+         cpucluster:
+         - name: <coreA>
+         - name: <coreB>
+           ...
+       - name: <soc2>
+     - name: <series-2-name>
+       ...
+
+
+Write your devicetree
+*********************
+
+SoC devicetree include files are located in :file:`<zephyr-repo>/dts` folder
+under a corresponding :file:`<ARCH>/<VENDOR>`.
+
+The SoC :file:`dts/<ARCH>/<VENDOR>/<soc>.dtsi` describes your SoC hardware in
+the Devicetree Source (DTS) format and must be included by any boards which uses
+the SoC.
+
+If a highlevel :file:`<arch>.dtsi` file exists, then a good starting point is to
+include this file in your :file:`<soc>.dtsi`.
+
+In general, :file:`<soc>.dtsi` should look like this:
+
+.. code-block:: devicetree
+
+   #include <arch>/<arch>.dtsi
+
+   / {
+   	chosen {
+   		/* common chosen settings for your SoC */
+   	};
+
+   	cpus {
+   		#address-cells = <m>;
+   		#size-cells = <n>;
+
+   		cpu@0 {
+   		device_type = "cpu";
+   		compatible = "<compatibles>";
+   		/* ... your CPU definitions ... */
+   	};
+
+   	soc {
+   		/* Your SoC definitions and peripherals */
+   		/* such as ram, clock, buses, peripherals. */
+   	};
+   };
+
+
+
+Multiple CPU clusters
+=====================
+
+Devicetree reflects the hardware. The memory space and peripherals available to
+one CPU cluster can be very different from another CPU cluster, therefore each
+CPU cluster will often have its own :file:`.dtsi` file.
+
+CPU cluster :file:`.dtsi` files should follow the naming scheme
+:file:`<soc>_<cluster>.dtsi`. A :file:`<soc>_<cluster>.dtsi` file will look
+similar to a SoC :file:`.dtsi` without CPU clusters.
+
+Write Kconfig files
+*******************
+
+Zephyr uses the Kconfig language to configure software features. Your SoC
+needs to provide some Kconfig settings before you can compile a Zephyr
+application for it.
+
+Setting Kconfig configuration values is documented in detail in
+:ref:`setting_configuration_values`.
+
+There is one mandatory Kconfig file in the SoC directory, and two optional
+files for a SoC:
+
+.. code-block:: none
+
+   boards/<vendor>/plank
+   ├── Kconfig.soc
+   ├── Kconfig
+   └── Kconfig.defconfig
+
+:file:`Kconfig.soc`
+  A shared Kconfig file which can be sourced both in Zephyr Kconfig and sysbuild
+  Kconfig trees.
+
+  This file selects the SoC family and series in the Kconfig tree and potential
+  other SoC related Kconfig settings. In some cases a SOC_PART_NUMBER.
+  This file must not select anything outside the re-usable Kconfig SoC tree.
+
+  A :file:`Kconfig.soc` may look like this:
+
+  .. code-block:: kconfig
+
+     config SOC_<SOC_NAME>
+     	bool
+     	select SOC_SERIES_<series name>
+
+     config SOC
+     	default "SoC name" if SOC_<SOC_NAME>
+
+  Notice that ``SOC_NAME`` is a pure upper case version of the SoC name.
+
```

Comment:
note to self: remember to highlight that the SoC name must be in sync with the value defined in `soc.yml`

Reply:
Done